onnxruntime/test/framework/ort_model_only_test.cc

```diff
@@ -237,6 +237,33 @@ static void DumpOrtModelAsJson(const std::string& model_uri) {
 }
 */
 
+TEST(OrtModelOnlyTests, SerializeMnistToOrtFormat) {
```

Comment:
What's the value of this test? Is it there some functionality it is covering that the existing tests don't?

Reply:
I ran into errors while running fuzz test using the MNIST example where the other testcase ran okay. You have provided me the fix and then MNIST worked fine. Because this example uncovered a problem. I thought that it is a good idea to keep the Mnist test.

---
In reply to: [521676385](https://github.com/microsoft/onnxruntime/pull/5771#discussion_r521676385) [](ancestors = 521676385)

Reply:
If you remove the change to inference_session.cc does this test fail? 

IIRC that required the ORT model being created with an optimization level of 2, and that model being loaded in a full build where we perform level 3 optimizations which change some nodes, but attempt to use the kernel hashes from the ORT model. I didn't notice the necessary steps here to replicate that scenario though.

The test should be commented to explain exactly what is being tested if it's for that edge case. Otherwise the test is likely to be removed as it's unclear what value is being added. I'd suggest the test name shouldn't be about 'mnist' but instead about loading an ORT format model in a full build where additional optimizations may be applied. 

Does LoadMnistOrtFormatModel test add value? The 'load only' tests are all we can run in a minimal build, so no optimizers would run in that case.

---
In reply to: [521713904](https://github.com/microsoft/onnxruntime/pull/5771#discussion_r521713904) [](ancestors = 521713904,521676385)

Reply:
Yes. It did fail when I removed the change to inference_sesion.cc

---
In reply to: [521808460](https://github.com/microsoft/onnxruntime/pull/5771#discussion_r521808460) [](ancestors = 521808460,521713904,521676385)

Reply:
I think it is better make the flow independent of optimization settings. With your fix in inference_session.cc it is working now without choosing optimization level 2. This test case is to cover the scenario as the other model, ort_github_issue_4031.onnx.ort, did not catch the issue.

---
In reply to: [521808460](https://github.com/microsoft/onnxruntime/pull/5771#discussion_r521808460) [](ancestors = 521808460,521713904,521676385)

Reply:
ValidateOrtFormatModelDoesNotRunOptimizersInFullBuild

---
In reply to: [522446895](https://github.com/microsoft/onnxruntime/pull/5771#discussion_r522446895) [](ancestors = 522446895,521808460,521713904,521676385)